src/_N001_Email.cpp

```diff
@@ -473,9 +472,14 @@ bool NPlugin_001_MTA(WiFiClient& client, const String& aStr, uint16_t aWaitForPa
     String line;
     safeReadStringUntil(client, line, '\n', 1024, timeout);
 
-    line.replace("-", " "); // Must Remove optional dash from MTA response code.
+    // response could be like: '220 domain', '220-domain','220+domain'

```

Comment:
I think this is the main issue here...

Reply:
This one was unreadable for me ;)
Could add replace + to space but it will be more unreadable. 

In my opinion this flush created problem

Fought something is wrong with that special fix for lagging server but didn't dive int it at that time. 

Please create build to test for user i don't know why can't download any file from gitpod.

Reply:
Build is running.
And yes the flush() is also creating havoc everywhere.
It is the kind of "fix" which keeps delivering.
Maybe we should remove the flush also from this email plugin.

Reply:
If you want to chat with server, especially send command and wait for response, you should flush received data to be sure you receive response for this command. 
In fact this plugin should be converted to state machine.

Reply:
Yep, totally agree on the state machine.
As should be beneficial for lots of other code in ESPEasy (*kuch* WiFi *kuch*) as it also makes the code non-blocking and much easier to maintain.

For another project I recently made some tool to scan comments in code to collect and create a .dot file to generate PDFs with the statemachine.
I'm thinking of adding something similar to ESPEasy so we can always have a clear chart of state machines wherever they are used and have them up-to-date.
This really eases finding bugs.

Reply:
@TD-er  maybe you can cherry pick these commits to mega branch